units/include/scipp/units/unit.h

```diff
@@ -5,97 +5,19 @@
-make_unit(const std::tuple<Ts...> &, const std::tuple<Extra...> &) {
-  return {};
-}
-
-template <class Unit, class... Units>
-constexpr scipp::index unit_index(Unit unit, std::tuple<Units...>) {
-  // We cannot rely on matching *types* in boost units so a simple lookup of
-  // `Unit` in the tuple of `Units` would often not succeed. Instead, we need to
-  // use the comparison operator.
-  constexpr auto match =
-      std::array<bool, sizeof...(Units)>{(unit == Units{})...};
-  for (size_t i = 0; i < sizeof...(Units); ++i)
-    if (match[i])
-      return i;
-  return -1;
-}
-
-} // namespace detail
-
-namespace boost_units {
-using nanosecond = decltype(boost::units::si::nano * boost::units::si::seconds);
-using millimeter = decltype(boost::units::si::milli * boost::units::si::meter);
-static constexpr boost::units::si::dimensionless dimensionless;
-static constexpr boost::units::si::length m;
-static constexpr boost::units::si::time s;
-static constexpr boost::units::si::mass kg;
-static constexpr boost::units::si::temperature K;
-static constexpr boost::units::si::plane_angle rad;
-static constexpr nanosecond ns;
-static constexpr millimeter mm;
-static constexpr decltype(boost::units::degree::plane_angle{} *
-                          dimensionless) deg;
-
-struct supported_units {
-  using type = decltype(detail::make_unit(
-      std::make_tuple(m, dimensionless / m, us, dimensionless / us, angstrom,
-                      dimensionless / angstrom),
-      std::make_tuple(
-          dimensionless, rad, deg, rad / deg, deg / rad, counts,
-          dimensionless / counts, s, kg, meV, dimensionless / s, counts / us,
-          counts / angstrom, counts / meV, m *m *m, meV *us *us / (m * m),
-          meV *us *us *dimensionless, kg *m / s, m / s, c, c *m, meV / c,
-          dimensionless / c, K, us / angstrom, us / (angstrom * angstrom),
-          us / (m * angstrom), angstrom / us, m *angstrom, (m * angstrom) / us,
-          ns, mm, dimensionless / meV, rad / s, (m * m) / s)));
-};
-struct counts_unit {
-  using type = decltype(boost_units::counts);
-};
-} // namespace boost_units
-using supported_units_t = typename boost_units::supported_units::type;
-using counts_unit_t = typename boost_units::counts_unit::type;
 
 class SCIPP_UNITS_EXPORT Unit {
 public:
   constexpr Unit() = default;
-  template <class Dim, class System, class Enable>
-  explicit constexpr Unit(boost::units::unit<Dim, System, Enable> unit) {
-    constexpr auto index = detail::unit_index(unit, supported_units_t{});
-    static_assert(index >= 0, "Unsupported unit.");
-    m_index = index;
-  }
-  static constexpr Unit fromIndex(const int64_t index) {
-    Unit u;
-    u.m_index = index;
-    return u;
-  }
+  constexpr Unit(const llnl::units::precise_unit &u) noexcept : m_unit(u) {}
+  Unit(const std::string &unit);
 
-  constexpr scipp::index index() const noexcept { return m_index; }
+  constexpr auto underlying() const noexcept { return m_unit; }
```

Comment:
Could this be made private and operator level support provided through `Unit` directly, such as `operator * (...)` This would better encapsulate the use of the underlying unit library/type